.github/workflows/test.yml

```
@@ -8,16 +8,22 @@ on:
       - "main"
       - "rc"
       - "hotfix-rc-*"
-  pull_request:
+  pull_request_target:
+    types: [opened, synchronize]
 
 defaults:
   run:
     shell: bash
 
 jobs:
+  check-run:
+    name: Check PR run
+    uses: bitwarden/gh-actions/.github/workflows/check-run.yml@main
+
   test:
     name: Run tests
     runs-on: ubuntu-22.04
+    needs: check-run
```

🎨 Let's also assert some token permissions here.

```suggestion
    needs: check-run
    permissions:
      contents: read
      pull-requests: write
```

---

@withinfocus leaving a reference to [assigning-permissions-to-jobs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) 

`contents: read` permits an action to list the commits
`pull-requests: write` permits an action to add a label to a pull request